### DISCUSSION TITLE
可以考虑下增加数据的持久化吗，需求量不大

### DISCUSSION THREAD

**Discussion by LinuxSuRen** [zeabur/.github#28] [Ideas]
您好，我的一个[接口开发、测试工具](https://api-testing.zeabur.app/)已经部署在咱们平台上了，但发现没有设置持久化的地方。我这个应用只需要很少量的存储，100M 足够。

**Comment by yuaanlin** [comment]
您好，Zeabur 是支持数据持久化的 https://zeabur.com/docs/zh-CN/deploy/customize-prebuilt#各栏位介绍

**Reply by LinuxSuRen** [reply]
Free Plan 没有，对吗

**Reply by yuaanlin** [reply]
对的

**Reply by LinuxSuRen** [reply]
另外，只有 PreBuilt 才能持久化，感觉有点不合理啊。基于代码仓库的也有持久化需求啊。而且，这个不是 Free Plan 或者付费的问题。

**Reply by yuaanlin** [reply]
对于 git 类型的服务，Zeabur 支持滚动发布（rolling update），让你 push 代码以后网站可以在持续可用的情况下发布新版本。

如果您在 git 服务绑定了一个持久化磁盘，将导致滚动发布失败，因为同时两个实例对同一份持久化数据进行读写，将导致数据损毁。

**Reply by LinuxSuRen** [reply]
直接用镜像部署和 Git 仓库构建出镜像后部署，不是同样的存在这个问题吗？这确实是个滚动发布的问题。

**Reply by yuaanlin** [reply]
建议您将需要持久化的数据保存在数据库或对象存储服务中，让 git 服务保持无状态的设计

**Reply by LinuxSuRen** [reply]
> 两个实例对同一份持久化数据进行读写，将导致数据损毁

这个我明白。其实不太理解的是，如果 PreBuilt 情况下，如果不是一个无服务，同样存在你的说的这个问题

**Reply by yuaanlin** [reply]
prebuilt 服务目前 Zeabur 是不支持滚动发布的，若您手动更新镜像版本将会有短暂的重启

**Reply by LinuxSuRen** [reply]
就是说， prebuilt 不会存在多示例同时运行以及读写数据的，对吗？

**Reply by yuaanlin** [reply]
> 就是说， prebuilt 不会存在多示例同时运行以及读写数据的，对吗？

是的

**Reply by LinuxSuRen** [reply]
感谢耐心解答。

**Comment by yuaanlin** [comment]
@LinuxSuRen 后续有其他问题的话欢迎到 Zeabur 官方 Discord 技术支持频道 https://discord.zeabur.com/ 发帖～
